src/trade/test/TradeParamsSelector.js

```diff
@@ -0,0 +1,89 @@
+import {
+    currencySelector,
+    tradeProposalInfoSelector,
+    tradePurchaseInfoSelector,
+    tradesErrorSelector,
+    tradesUIStatesSelector,
+} from '../../_store/directSelectors';
+import { createSelector, createStructuredSelector } from 'reselect';
+import findDeep from 'binary-utils/lib/findDeep';
+import filterObjectBy from 'binary-utils/lib/filterObjectBy';
+import { assetsIsOpenSelector, availableContractsSelector } from '../TradeSelectors';
+import { paramPerTrade, pipSizePerTrade, tradeViewChartPerTrade } from './TradeViewChartSelector';
+import { mockedContract } from '../../_constants/MockContract';
+
+const contractPerTrade = index => createSelector(
+    [availableContractsSelector, paramPerTrade(index)],
+    (contracts, param) => {
+        const symbol = param.get('symbol');
+        return contracts.get(symbol);
+    }
+);
+
+const marketIsOpenPerTrade = index => createSelector(
+    [assetsIsOpenSelector, paramPerTrade(index)],
+    (assetsIsOpen, param) => {
+        const symbol = param.get('symbol');
+        return assetsIsOpen[symbol].isOpen;
+    }
+);
+
+const errorPerTrade = index => state => tradesErrorSelector(state).get(index);
+const proposalPerTrade = index => state => tradeProposalInfoSelector(state).get(index);
+const uiStatePerTrade = index => state => tradesUIStatesSelector(state).get(index);
+const purchasePerTrade = index => state => tradePurchaseInfoSelector(state).get(index);
+
+const getStartLaterOnlyContract = contract => {
+    const startLaterCategories =
+        filterObjectBy(contract, child =>
```

Comment:
Hmm, I cannot visualize it directly, let me think a bit, but yea, I think we under-utilize the Record type provided by immutableJS (I know it exists from your commit XD)